src/Dao/CategoryDao.php

```diff
@@ -10,40 +10,51 @@
 namespace OxidEsales\GraphQL\Example\Dao;
 
 use OxidEsales\EshopCommunity\Internal\Framework\Database\QueryBuilderFactoryInterface;
+use OxidEsales\GraphQL\Base\Service\LegacyServiceInterface;
 use OxidEsales\GraphQL\Example\DataObject\Category;
+use OxidEsales\GraphQL\Example\Exception\CategoryNotFoundException;
 
 class CategoryDao implements CategoryDaoInterface
 {
 
     /** @var QueryBuilderFactoryInterface $queryBuilderFactory */
     private $queryBuilderFactory;
 
+    /** @var LegacyServiceInterface */
+    private $legacyService;
+
     public function __construct(
-        QueryBuilderFactoryInterface $queryBuilderFactory
+        QueryBuilderFactoryInterface $queryBuilderFactory,
+        LegacyServiceInterface $legacyService
     ) {
         $this->queryBuilderFactory = $queryBuilderFactory;
+        $this->legacyService = $legacyService;
     }
 
-    public function getCategoryById(string $id, int $languageId, int $shopId): ?Category
+    /**
+     * @param string $id
+     * @param int $languageId
+     * @return Category
+     * @throws CategoryNotFoundException
+     */
+    public function getCategoryById(string $id, int $languageId): ?Category
     {
         $queryBuilder = $this->queryBuilderFactory->create();
         $title = $languageId === 0 ? "OXTITLE" : "OXTITLE_$languageId";
 
         $queryBuilder->select(['OXID', $title, 'OXPARENTID'])
                      ->from('oxcategories')
                      ->where($queryBuilder->expr()->andX(
-                         $queryBuilder->expr()->eq('OXID', ':oxid'),
-                         $queryBuilder->expr()->eq('OXSHOPID', ':shopid')
+                         $queryBuilder->expr()->eq('OXID', ':oxid')
                      ))
-                     ->setParameter('oxid', $id)
-                     ->setParameter('shopid', $shopId);
+                     ->setParameter('oxid', $id);
         $result = $queryBuilder->execute();
         if (!$result instanceof \Doctrine\DBAL\Driver\Statement) {
-            return null;
+            throw new CategoryNotFoundException("Error executing category query.");
         }
         $row = $result->fetch();
         if (!$row) {
-            return null;
+            throw new CategoryNotFoundException("There is no category with id '$id'.");
```

Comment:
I think returning `null` in case we can not find a category is perfectly valid, making this a `404` response in GraphQL should be implemented in the controller then. To me this is not an exceptional situation but just normal flow, thus i would not see the usage of this exception here.

```suggestion
            return null;
```